x64 version not reporting right version number with --list-devices

### Describe the bug

I have upgraded to 3.4.1 on ARM and x64 but the x64 under Ubuntu 22.04 does not report correctly ? yet ionos1 running debian 12 on x86 does work fine..
Both dess & cally are running ubuntu and the daemon log is correct but they report in the client as 3.4.0 ????



### Steps to reproduce

1. Upgraded to latest version
2. ran `sshnp -f @cconstab -t @ssh_1  --host @rv_am  --list-devices`



### Expected behavior

to report correct version

### Screenshots

Upgraded cally (both sshnp/sshnpd are upgraded) not that raksign1 is reporting correctly
```
cconstab@cally:~$ sshnp -t @ssh_1 -f @cconstab --list-devices -h @rv_am
Searching for devices...
Active Devices:
  cally - v3.4.0
  dess - v3.4.0
  ionos1 - v3.4.1
  raksign1 - v3.4.1
Inactive Devices:
  [X] No devices found
  ```
  
  but this in the logs 
  
  ```
  INFO|2023-08-23 18:24:08.090260|Monitor (@ssh_1)|monitor started for @ssh_1 with last notification time: null
INFO|2023-08-23 18:24:18.429434| sshnpd |Received: ping
INFO|2023-08-23 18:24:18.430299| sshnpd |ping received from @cconstab notification id : dfa6ff8f-b88f-457d-94f1-46eed9ee7561
INFO|2023-08-23 18:24:19.624180| sshnpd |SUCCESS:id: 1c918723-31ce-4ab3-9be6-99d04af482a1 status: NotificationStatusEnum.delivered for:  with value: {"devicename":"cally","version":"3.4.1"}
```
  

### Smartphones

```
cconstab@cally:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
cconstab@cally:~$
```

Same issue with sshnp upgrade

```
cconstab@ionos1:~$ bash -c "$(curl -fsSL https://getsshnp.noports.com)" -- --update
[ERROR] System failed to meet the following requirements:
  [X] Missing required dependency: sshd
cconstab@ionos1:~$ export PATH="$PATH:/usr/sbin"
cconstab@ionos1:~$ bash -c "$(curl -fsSL https://getsshnp.noports.com)" -- --update
Downloading sshnp from https://github.com/atsign-foundation/sshnoports/releases/download/v3.4.1/sshnp-linux-x64.tgz
mv: cannot overwrite non-directory '/home/cconstab/.atsign/temp//sshnp/sshnp' with directory '/home/cconstab/.atsign/temp//sshnp'
Installed binaries: sshnp at_activate sshrv

Update complete!
cconstab@ionos1:~$
```

Ok I had to add to the sshnpd@cconstab the `-u -s` flags now I get the right result but why was I getting 3.4.0 ??

```
cconstab@cally:~$ sshnp -f @cconstab -t @ssh_1  --host @rv_am  --list-devices
Searching for devices...
Active Devices:
  cally - v3.4.1
  dess - v3.4.0
  ionos1 - v3.4.1
  raksign1 - v3.4.1
  risc - v3.4.1
Inactive Devices:
  [X] No devices found
cconstab@cally:~$ cat .local/bin/sshnpd@cconstab
#!/bin/bash
# v2.0.0
# allow machine to bring up network
sleep 10
USER=$(whoami)
export USER
DEVICE="@ssh_1"
CLIENT="@cconstab"
NAME="cally"
ARGS="-v -u -s"
while true; do
  # shellcheck disable=SC2086
  "/home/cconstab"/.local/bin/sshnpd -a "$DEVICE" -m "$CLIENT" -d "$NAME" $ARGS
  sleep 10
done
cconstab@cally:~$
```

--- a/packages/sshnoports/lib/sshnpd/sshnpd_impl.dart
+++ b/packages/sshnoports/lib/sshnpd/sshnpd_impl.dart
@@ -159,8 +159,8 @@ class SSHNPDImpl implements SSHNPD {
         );
 
     // Refresh the device entry now, and every hour
+    await _refreshDeviceEntry();
     if (makeDeviceInfoVisible) {
-      await _refreshDeviceEntry();
       Timer.periodic(
         const Duration(hours: 1),
         (_) async => await _refreshDeviceEntry(),
@@ -801,10 +801,6 @@ class SSHNPDImpl implements SSHNPD {
 
   /// This function creates an atKey which shares the device name with the client
   Future<void> _refreshDeviceEntry() async {
-    if (!makeDeviceInfoVisible) {
-      return;
-    }
-
     const ttl = 1000 * 60 * 60 * 24 * 30; // 30 days
     var metaData = Metadata()
       ..isPublic = false
@@ -820,6 +816,20 @@ class SSHNPDImpl implements SSHNPD {
       ..sharedWith = managerAtsign
       ..namespace = SSHNPD.namespace
       ..metadata = metaData;
+
+    if (!makeDeviceInfoVisible) {
+      logger.info('Deleting old device info for $device (if it exists)');
+      try {
+        await atClient.delete(
+          atKey,
+          deleteRequestOptions: DeleteRequestOptions()
+            ..useRemoteAtServer = true,
+        );
+      } catch (e) {
+        stderr.writeln(e.toString());
+      }
+      return;
+    }
 
     try {
       logger.info('Updating device info for $device');

